kpi/tests/test_assets.py

```diff
@@ -1002,3 +1003,40 @@ def test_asset_is_excluded_from_projects_list_flag(self):
         # Fetch the newly created asset and verify the flag is False
         owner_asset = Asset.objects.get(name='Breakfast')
         self.assertFalse(owner_asset.is_excluded_from_projects_list)
+
+    def test_asset_visibility_after_transfer(self):
+        """
+        Test to ensure that an asset shared with an organization owner remains
+        visible after the asset owner joins a different organization
+        """
+        # Step 1: Create an asset owned by an external user
+        self._create_asset_by_bob()
+        asset = Asset.objects.get(owner=self.external_user)
+
+        # Step 2: External user shares the asset explicitly with OrgA's owner
+        asset.assign_perm(self.org_owner, PERM_VIEW_ASSET)
+        self.assertTrue(self.org_owner.has_perm(PERM_VIEW_ASSET, asset))
+
+        # Step 3: Verify that OrgA's owner can see the asset in their asset list
+        self.client.force_login(self.org_owner)
+        response = self.client.get(self.asset_list_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(
+            any(obj['uid'] == asset.uid for obj in response.data['results'])
+        )
+
+        # Step 4: Invite the external user to another organization (OrgB)
+        another_org = Organization.objects.create(
+            id='org1234', name='Another Organization', mmo_override=True
+        )
+        another_org.add_user(self.thirduser)
+
+        self.client.force_login(self.org_owner)
+        self._add_user_to_organization(self.external_user, another_org)
+
+        # Step 5: Verify that the asset is still visible in OrgA and OrgB
```

Comment:
I suppose that, here, we want to test (only) that OrgB owner can see the asset. We already test the visibility for OrgA previously? Please update your comment. Otherwise, something is missing here.
We are still logged as OrgA owner and we only test the response of the "MyProjects" list.
OrgA owner should see the project in their "MyProjects" list, but OrgB owner should see the project in their "My Org Projects" list.